### PR TITLE
Profit tab

### PR DESCRIPTION
**Pull request by isthiscodefluffy**
Here is the pirate's game tab. Hopefully it will find itself a home here.

### CODE REVIEW

**Comment by jaghaimo** [comment]
Please fix conflicts and pipeline :)

File: src/stelnet/board/commodity/table/profit/ProfitTableContent.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        Collections.sort(rows);
+
+        return rows;
+    }
+
+    private List<TableProfitRow> getViableSellMarketsForBuy(MarketAPI buyMarket) {
+        List<TableProfitRow> rows = new ArrayList<>();
+
+        for (MarketAPI sellMarket : sellmarkets) {
+            if (ProfitCalculator.getPotentialProfit(buyMarket, sellMarket, commodityId) <= 100000) {
```
Comment (jaghaimo) [inline src/stelnet/board/commodity/table/profit/ProfitTableContent.java:54]: Magic number. Use `ProfitCalculator.MIN_PROFIT_VALUE`

File: src/stelnet/board/commodity/table/profit/ProfitCalculator.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            return 0;
+        }
+
+        CommodityOnMarketAPI buyFromCommodity = buyMarket.getCommodityData(commodityId);
+        CommodityOnMarketAPI sellToCommodity = sellMarket.getCommodityData(commodityId);
+
+        int available = TableCellHelper.getAvailable(buyFromCommodity);
+        int demand = TableCellHelper.getDemand(sellMarket, sellToCommodity);
+
+        if (available < 400) {
```
Comment (jaghaimo) [inline src/stelnet/board/commodity/table/profit/ProfitCalculator.java:25]: Magic number. Use `MIN_REQUIRED_VOLUME` instead.

File: src/stelnet/board/commodity/table/profit/ProfitCalculator.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package stelnet.board.commodity.table.profit;
+
+import com.fs.starfarer.api.campaign.econ.CommodityOnMarketAPI;
+import com.fs.starfarer.api.campaign.econ.MarketAPI;
+import stelnet.board.commodity.price.Price;
+import stelnet.board.commodity.price.SupplyPrice;
+import stelnet.util.TableCellHelper;
+
+public class ProfitCalculator {
```
Comment (jaghaimo) [inline src/stelnet/board/commodity/table/profit/ProfitCalculator.java:9]: Perhaps `ProfitHelper`?

**Comment by jaghaimo** [comment]
Could you try also changing `prettier.yml` GitHub action? Replace the existing Checkout stage with this one:

```
            - name: Checkout
              uses: actions/checkout@v2
              with:
                  lfs: true
```

File: src/stelnet/board/commodity/table/profit/ProfitCalculator.java (hunk 3 above)
Comment (isthiscodefluffy) [reply src/stelnet/board/commodity/table/profit/ProfitCalculator.java:9]: Having "Helper" or "Util" in your class name abstracts away the actual intent and functionality of the class. What ends up happening is that, you will have Helper classes that start violating Single Responsibility Principle. If the name would describe the precise goal of a class then your inner programmer-senses would guard you against violating that principle. If the name is generic then the classes become generic, which in turn leads to badly managed code.

**Pull request merged**